packages/components/src/components/domainproperties/dataset/actions.ts

```diff
@@ -23,40 +31,92 @@ import {
     DATASET_NAME_TIP,
     DATASPACE_TIP,
     TAG_TIP,
+    TIME_KEY_FIELD_DISPLAY,
+    TIME_KEY_FIELD_KEY,
     VISIT_DATE_TIP,
 } from './constants';
 
-export const fetchCategories = async () => {
-    // TODO: Replace this with server side call
-    return {
-        categories: [
-            { label: 'A', value: 20 },
-            { label: 'B', value: 21 },
-            { label: 'C', value: 22 },
-        ],
-    };
-};
-
-export const fetchCohorts = async () => {
-    // TODO: Replace this with server side call
-    return {
-        cohorts: [
-            { label: 'Cohort1', value: 1 },
-            { label: 'Cohort2', value: 2 },
-            { label: 'Cohort3', value: 3 },
-        ],
-    };
-};
-
-export const fetchVisitDateColumns = async () => {
-    // TODO: Keeping this action until next story in which visitDateColumns will be pulled from state change (for date fields) in the Domain Form.
-    return {
-        visitDateColumns: [
-            { label: 'Date', value: 'date' },
-            { label: 'Arrival Date', value: 'arrivalDate' },
-        ],
-    };
-};
+export function fetchCategories(): Promise<List<Option>> {
+    return new Promise((resolve, reject) => {
+        selectRows({
+            saveInSession: true,
```

Comment:
Do we need this saveInSession param? Usually that is used if you are making an executeSql call and then want to re-query that result by temp query name.